src/ui/groups_widgets.cc

```diff
@@ -790,34 +797,82 @@ void DictGroupsWidget::addAutoGroupsByFolders()
     return;
   }
 
-  QMultiMap< QString, sptr< Dictionary::Class > > dictMap;
+  // Map from dict to ContainerFolder's parent
+  QMap< sptr< Dictionary::Class >, QString > dictToContainerFolder;
 
   for ( const auto & dict : *activeDicts ) {
-    QString dictFolder = dict->getContainingFolder();
+    QDir c = dict->getContainingFolder();
 
-    if ( dictFolder.isEmpty() ) {
-      continue;
+    if ( !c.cdUp() ) {
+      cdUpWentWrong( c.absolutePath() );
+      return;
     }
```

Comment:
continue other than `return` should be better

Reply:
message box is too intrusiving .  can be ignored and continue to process other dictionaries.
and all these dictionaries can be put in a different group such as `other` or `unclassified`

Reply:
If we continue here, then the grouping result will be incomplete. A user won't know why some dictionaries won't showing up.

The cdUp fails only happens when a user put the folder under root dir like `/`. We should tell the user what's wrong and let the user move the folder and do it again.

Reply:
![image](https://user-images.githubusercontent.com/20123683/234550149-edfde44d-f9d4-4d74-9e61-ae43e778c1fc.png)

I think intrusiveness is acceptable :sweat_smile:

Reply:
then users can only use this feature under a strict condition.
If this is designed,this is ok. wait for user feedback

Reply:
I think if someone wants to use this feature, and he does not put the dict files' container folder under a directory with a name (aka not `/`), then GD should not behave like nothing wrong happens.